kratos/utilities/multipoint_constraint.h

```diff
@@ -0,0 +1,582 @@
+//    |  /           |

```

Comment:
@RiccardoRossi would you please review this one?

Reply:
I took a look and it looks correct to me. However i cannot say i understand all the details of the implementation. 

@adityaghantasala can u refer to some documentation about the algorithm? (i mean in the comments)